apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -658,6 +675,15 @@ private void mandatoryMissing(ApikeyAction apikeyUpdate) throws ApikeyException
             throw new ApikeyException(400, BAD_EMAIL_FORMAT, BAD_EMAIL_FORMAT);
         }
     }
-
+    private void apiKeyAlreadyExist(ApikeyAction apikeyUpdate) throws ApikeyException{
+        Apikey apikey= this.apikeyRepo.findByEmailAndAppName(apikeyUpdate.getEmail(), apikeyUpdate.getAppName());
+        System.out.println(apikey);
+        if(apikey!=null){
+            String message = APIKEYALREADYEXIST +" for : " +apikeyUpdate.getEmail() +", " +apikeyUpdate.getAppName();
+            throw new ApikeyException(400, APIKEYALREADYEXIST, message);
```

Comment:
More efficient to just return apikey != null and use that value to send the correct response